framework/sandstone.cpp

```diff
@@ -2423,6 +2423,27 @@ static void wait_delay_between_tests(bool skip_wait)
     usleep(useconds);
 }
 
+//we take here walue from delay_between_tests and we make it deviate in given range
+// i.e +/- 10%.
+static void wait_delay_between_tests_with_deviation(double deviation_percentage, bool skip_wait)
+{
+    double random_factor = 0.0;
+    useconds_t sleep_time = 0;
+
+    if (!skip_wait) {
+        random_factor = frandom_scale(2.0) - 1.0; // get a random number from -1.0 to 1.0
+        random_factor = random_factor * deviation_percentage / 100;
+        Duration deviation =
+            std::chrono::duration_cast<std::chrono::microseconds>(sApp->delay_between_tests * random_factor);
+        sleep_time =
+            std::chrono::duration_cast<std::chrono::microseconds>(sApp->delay_between_tests + deviation).count();
+    }       
+
+    logging_printf(LOG_LEVEL_VERBOSE(1), "Sleep time %" PRId32"\n", sleep_time);
```

Comment:
This line needs to start with a hash mark so it's a comment on the YAML output. I'd also increase it to level 2.